src/Builder/CorrelationIdsRegistryBuilder.php

```diff
@@ -56,7 +56,7 @@ public function buildFromRequestHeaders(array $requestHeaders): CorrelationIdsRe
         return $this->build(
             $this->extractRequestHeader(
                 $sanitizedRequestHeaders,
-                $this->sanitize($this->provider->provideParentCorrelationIdHeaderName())
+                $this->sanitize($this->provider->provideCurrentCorrelationIdHeaderName())
```

Comment:
This is wrong, if we merge this, we break everything :)

The role of this builder class is to build a registry, by providing optionally parent and root correlation ids if available. Each new build registry MUST HAVE a new correlation id (new actor => new own id). The switch you want to perform is actually MUST NOT be done here, but in libraries using this core.

Ex: App A calls App B via guzzle with that middleware enabled: https://github.com/oat-sa/lib-correlation-ids-guzzle/blob/master/src/Middleware/CorrelationIdsGuzzleMiddleware.php#L45

=> here, you see that the switch is done for you (you don’t have to care about it): 
- App A used the registry builder (got his own new current correlation id)
- App A calls app B by providing as parent id its own current id
- App B will create its registry with its own current id, and by referencing the app A id as the parent one

Please recheck the principles here: https://github.com/oat-sa/lib-correlation-ids#principles

I decline the PR.

Reply:
Thanks for the explanation. My main problem when doing things from outside was the abstraction of the nameProvider, and doing it in the middleware is obviously the solution.